home/migrations/0006_auto_20200623_2207.py

```diff
@@ -0,0 +1,254 @@
+# Generated by Django 3.0.5 on 2020-06-24 05:07
+
+from django.db import migrations, models
+import django.db.models.deletion
+import django.utils.timezone
+import uuid
+
+
+def migrate_appusers(apps, schema_editor):
+    with schema_editor.connection.cursor() as cursor:
+        # create unique accounts for each email
+        cursor.execute("""
+            INSERT INTO home_account (email, name, zip, age, created, updated)
+            SELECT au.email, au.name, au.zip, au.age, au.created, NOW()
+            FROM home_appuser au
+            WHERE au.created=(
+                SELECT MIN(au2.created)
+                FROM home_appuser au2
+                WHERE LOWER(au2.email)=LOWER(au.email)
+            )
+        """)
+        # copy all of the device ids into the new devices table
+        cursor.execute("""
+            INSERT INTO home_device (device_id, account_id, created)
+            SELECT au.account_id, ac.id, au.created
+            FROM home_appuser au
+            JOIN home_account ac
+            ON LOWER(au.email)=LOWER(ac.email)
+        """)
+        # create some dummy rows to set initial default values in migration
+        cursor.execute("""
+            INSERT INTO home_account (id, email, name, zip, age, created, updated)
+            VALUES (0, 'dummy@email.com', 'Dummy Account', '94103', 0, NOW(), NOW())
+        """)
+        cursor.execute("""
+            INSERT INTO home_device (device_id, account_id, created)
+            VALUES ('0', 0, NOW())
+        """)
+
+
+def migrate_walks(apps, schema_editor):
+    with schema_editor.connection.cursor() as cursor:
+        # update daily walk records
+        cursor.execute("""
+            UPDATE home_dailywalk
+            SET device_id=sq.device_id, account_id=sq.account_id
+            FROM (
+                SELECT dev.device_id device_id, dev.account_id account_id, au.account_id appuser_id
+                FROM home_device dev
+                JOIN home_appuser au ON au.account_id=dev.device_id
+            ) AS sq
+            WHERE home_dailywalk.appuser_id=sq.appuser_id
+        """)
+        # delete duplicate records in daily walks, keeping the most recent
+        cursor.execute("""
+            DELETE FROM home_dailywalk dw
+            WHERE dw.updated<>(
+                SELECT MAX(dw2.updated)
+                FROM home_dailywalk dw2
+                WHERE dw2.account_id=dw.account_id
+                AND dw2.date=dw.date
+            )
+        """)
+        # update intentional walk records
+        cursor.execute("""
+            UPDATE home_intentionalwalk
+            SET device_id=sq.device_id, account_id=sq.account_id
+            FROM (
+                SELECT dev.device_id device_id, dev.account_id account_id, au.account_id appuser_id
+                FROM home_device dev
+                JOIN home_appuser au ON au.account_id=dev.device_id
+            ) AS sq
+            WHERE home_intentionalwalk.appuser_id=sq.appuser_id
+        """)
```

Comment:
Looking at this, I forgot to delete the dummy Account/Device rows after migrating the data. I'll add that in tomorrow...